src/Transport/Addressing/Sanitization/Strategies/EndpointOrientedTopologySanitization.cs

```diff
@@ -4,21 +4,13 @@
 
     public class EndpointOrientedTopologySanitization : ISanitizationStrategy
     {
-        IValidationStrategy validationStrategy;
-
-        public EndpointOrientedTopologySanitization(IValidationStrategy validationStrategy)
-        {
-            this.validationStrategy = validationStrategy;
-        }
-
-        public string Sanitize(string entityPathOrName, EntityType entityType)
+        public string Sanitize(string entityPathOrName, EntityType entityType, ValidationResult validationResult)
         {
             // remove invalid characters
             var rgx = new Regex(@"[^a-zA-Z0-9\-\._]");
```

Comment:
Not quite the same...
`EntityNameValidationV6Rules`:              "^[0-9A-Za-z_\.\-]+$"
`EndpointOrientedTopologySanitization`: "[^a-zA-Z0-9\-\._]"

`EntityNameValidationV6Rules` uses the regex to validate characters (with requirement of length more than zero, "+" at the end)
Regex in `EndpointOrientedTopologySanitization` represents all the characters that are not legal. Hence the "^" in the beginning of the character group.